core/src/main/java/hudson/PluginManager.java

```diff
@@ -1213,8 +1213,13 @@ protected PluginStrategy createPluginStrategy() {
 			LOGGER.info("Falling back to ClassicPluginStrategy");
 		}
 
-		// default and fallback
-		return new ClassicPluginStrategy(this);
+		// Default and fallback strategy
+        // The default Plugin Manager implementation does not propagate/log stacktraces for ClassNotFoundExceptions,
+        // hence stacktraces are omitted by default for better performance.
+        boolean omitClassNotFoundExceptionStacktraces =SystemProperties.getBoolean(
+                ClassicPluginStrategy.class.getName() + ".omitClassNotFoundStacktraces", true);
```

Comment:
Conventionally this would be a non`final` `static` constant.

Reply:
There are other system properties which are resolved on-demand. Taking the specifics of Jenkinsfile Runner codebase, I would prefer to keep more deterministic initialization than static initializer

Reply:
Again, why would you need to do anything at all in JFR if the default is set to omit stack traces, which it should be?